### DISCUSSION TITLE
Laravel doesn't give the correct locale in Provider

### DISCUSSION THREAD

**Discussion by mstfkhazaal** [laravel/framework#46706] [Q&A]
### Laravel Version

10.6.1

### PHP Version

8.2.4 

### Database Driver & Version

Mysql  Ver 8.0.32-0ubuntu0.22.04.2 for Linux on x86_64 ((Ubuntu))

### Description

Laravel doesn't give the correct locale in Any Provider and session is empty

### Steps To Reproduce

I use app() -> getLocale()
When the site is fully loaded it gives me the correct language from session
but in Povider i get empty session and get the default language of my website

**Comment by henzeb** [comment]
Service providers come before sessions. If you need to do something in a request, do so in a Middleware.

What exactly are you trying to do that you need locale for in a service provider?
